homeassistant/components/squeezebox/media_player.py

```diff
@@ -587,4 +591,25 @@ async def async_browse_media(self, media_content_type=None, media_content_id=Non
             "search_id": media_content_id,
         }
 
-        return await build_item_response(self._player, payload)
+        return await build_item_response(self, payload)
+
+    async def async_get_browse_image(self, browse_image):
+        """Get album art from Squeezebox server."""
+        if browse_image.startswith(self._internal_artwork_url):
+            return await self._async_fetch_image(browse_image, cache=False)
+        return (None, None)
+
+    def get_browse_image_url(self, browse_image):
+        """Generate an externally accessible url for a media browser image."""
+        base_url = get_url(self.hass, prefer_external=True)
+        thumbnail = browse_image
+        if thumbnail and thumbnail.startswith(self._internal_artwork_url):
+            thumbnail = (
+                f"{base_url}/api/media_player_proxy/{self.entity_id}"
+                f"?token={self.access_token}&browse_image={thumbnail}"
```

Comment:
We should not pass urls to the proxy. Instead pass the `media_content_id` and `media_content_type`. That way we will prevent integrations from accidentally creating a proxy that forwards any file from the internal network.